src/FSharpVSPowerTools.Logic/Setting.fs

```diff
@@ -1,42 +1,470 @@
 ﻿namespace FSharpVSPowerTools
 
+open System
+open System.Globalization
+open System.Runtime.CompilerServices
+open System.Xml
+open System.Xml.Linq
+open System.IO
+open System.Text
+open System.Collections.Generic
+open System.ComponentModel.Composition
+open EnvDTE
+open Microsoft.VisualStudio.Settings
+open Microsoft.VisualStudio.Shell
+open Microsoft.VisualStudio.Shell.Interop
+open Microsoft.VisualStudio.Shell.Settings
+open Microsoft.VisualStudio.Editor
+open System.Runtime.CompilerServices
+open FSharpVSPowerTools.Pervasive
+
+
+
+[<AutoOpen>]
+module Extensions =
+
+    let inline xelem  name (content:'a)  = new XElement(XName.Get name, content)
+
+    type XContainer with
+        member this.Element     name            = this.Element(XName.Get name)
+        member this.HasElement (name:string)    = isNotNull (this.Element name)
+        member this.AddElement  name            = this.Add (xelem name []); this.Element name
+
+        member this.GetOrCreateElement name =
+            if this.HasElement name then this.Element name else this.AddElement name
+
+        member this.SetOrCreateElement name value =
+            if this.HasElement name then (this.Element name).Value <- value 
+            else (this.AddElement name).Value <- value
+
+
+    type Dictionary<'key,'value> with
```

Comment:
we already have some Dictionaty related functions here https://github.com/fsprojects/VisualFSharpPowerTools/blob/master/src/FSharpVSPowerTools.Core/Utils.fs#L735